### PR TITLE
Allow named arguments for all dynamic pipeline steps

### LINKED ISSUE #125: Allow named arguments for all dynamic pipeline steps
**Describe the feature request**
Follow-up of #121 implementing the usage of named parameters for all dynamic pipeline steps.

**Upstream changes**
#124

**Additional context**
Update documentation and sample code in help files
Add more unit tests checking passed parameters

### PR DESCRIPTION
**Pull request by cpoenisch**
Resolves #125
Based on upstream #124

### CODE REVIEW

File: src/main/java/de/tracetronic/jenkins/plugins/ecutest/report/atx/pipeline/ATXServer.java [hunk 1]
```diff
@@ -85,6 +85,20 @@ public void publish(final boolean allowMissing, final boolean runOnFailed,
...
+     * Publishes ATX reports with given archiving settings as named arguments map.
+     *
+     * @param settings the settings map
+     */
+    @Whitelisted
+    public void publish(final Map<String, Object> settings) {
+        final Map<String, Object> stepVariables = Maps.newLinkedHashMap();
+        stepVariables.put("atxName", installation.getName());
+        stepVariables.put("atxInstallation", installation);
+        stepVariables.putAll(settings);
```
Comment (JanGu-TT) [inline src/main/java/de/tracetronic/jenkins/plugins/ecutest/report/atx/pipeline/ATXServer.java]: Should not we add also other named parameters: `allowMissing`, `runOnFailed`, `archiving`, `keepAll`

File: src/main/java/de/tracetronic/jenkins/plugins/ecutest/tool/pipeline/ETInstance.java [hunk 2]
```diff
@@ -161,6 +182,16 @@ public void startTS(final String toolLibsIniPath, final int tcpPort,
         script.invokeMethod("startTS", stepVariables);
     }
 
+    /**
+     * Stops ECU-TEST with given settings as named arguments map.
```
Comment (JanGu-TT) [inline src/main/java/de/tracetronic/jenkins/plugins/ecutest/tool/pipeline/ETInstance.java]: Stops Tool-Server

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/atx/ATXPublisher/help.html [hunk 3]
```diff
@@ -34,6 +37,10 @@
 def server = ATX.newServer('TEST-GUIDE', 'ECU-TEST')
 server.publish()
             </pre>
+            <pre>
+def server = ATX.newServer atxName: 'TEST-GUIDE', toolName: 'ECU-TEST'
+server.publish allowMissing: true, runOnFailed: true
```
Comment (JanGu-TT) [inline src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/atx/ATXPublisher/help.html]: Other parameters like:   `archiving` and `keepAll` are not documented in this example.

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/atx/ATXPublisher/help_de.html [hunk 4]
```diff
@@ -34,6 +37,10 @@
 def server = ATX.newServer('TEST-GUIDE', 'ECU-TEST')
 server.publish()
             </pre>
+            <pre>
+def server = ATX.newServer atxName: 'TEST-GUIDE', toolName: 'ECU-TEST'
+server.publish allowMissing: true, runOnFailed: true
```
Comment (JanGu-TT) [inline src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/atx/ATXPublisher/help_de.html]: Other parameters like: archiving and keepAll are not documented in this example.

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/generator/ReportGeneratorPublisher/help.html [hunk 5]
```diff
@@ -34,7 +37,11 @@
             </pre>
             <pre>
 def instance = ET.newInstallation('ECU-TEST', 'C:\\Program Files\\ECU-TEST 8.0')
-instance.publishGenerators([[name: 'JSON']], [name: 'Custom-JSON')])
+instance.publishGenerators([[name: 'JSON']], [[name: 'Custom-JSON']])
+            </pre>
+            <pre>
+def instance = ET.newInstallation toolName: 'ECU-TEST', installPath: C:\\Program Files\\ECU-TEST 8.0'
+instance.publishGenerators generators: [[name: 'JSON']], customGenerators: [[name: 'Custom-JSON']]
```
Comment (JanGu-TT) [inline src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/generator/ReportGeneratorPublisher/help.html]: Other parameters like: `allowMissing`, `runOnFailed`, `archiving` and `keepAll` are not documented in this example.

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/generator/ReportGeneratorPublisher/help_de.html [hunk 6]
```diff
@@ -37,6 +40,10 @@
 def instance = ET.newInstallation('ECU-TEST', 'C:\\Program Files\\ECU-TEST 8.0')
 instance.publishGenerators([[name: 'JSON']], [name: 'Custom-JSON')])
             </pre>
+            <pre>
+def instance = ET.newInstallation toolName: 'ECU-TEST', installPath: C:\\Program Files\\ECU-TEST 8.0'
+instance.publishGenerators generators: [[name: 'JSON']], customGenerators: [[name: 'Custom-JSON']]
```
Comment (JanGu-TT) [inline src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/generator/ReportGeneratorPublisher/help_de.html]: Other parameters like: `allowMissing`, `runOnFailed`, `archiving` and `keepAll` are not documented in this example.

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/generator/ReportGeneratorPublisher/help.html (hunk 5 above)
Comment (cpoenisch) [reply src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/generator/ReportGeneratorPublisher/help.html]: I mean these are just examples, full method syntax is provided above...

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/generator/ReportGeneratorPublisher/help_de.html (hunk 6 above)
Comment (cpoenisch) [reply src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/generator/ReportGeneratorPublisher/help_de.html]: s.a.

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/generator/ReportGeneratorPublisher/help.html (hunk 5 above)
Comment (cpoenisch) [reply src/main/resources/de/tracetronic/jenkins/plugins/ecutest/report/generator/ReportGeneratorPublisher/help.html]: Listed all archiving options in at least one example.

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/tool/StartTSBuilder/help.html [hunk 7]
```diff
@@ -36,6 +39,10 @@
 def instance = ET.newInstallation('ECU-TEST', 'C:\\Program Files\\ECU-TEST 8.0')
 instance.startTS()
             </pre>
+            <pre>
+def instance = ET.newInstallation toolName: 'ECU-TEST', installPath: C:\\Program Files\\ECU-TEST 8.0'
+instance.startTS toolLibsIniPath: C:\\ToolLibs.ini, tcpPort: 5017, timeout: '60', keepInstance: false
```
Comment (JanGu-TT) [inline src/main/resources/de/tracetronic/jenkins/plugins/ecutest/tool/StartTSBuilder/help.html]: timeout is small integer, would it work with quotas? `'60'`

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/tool/StartTSBuilder/help_de.html [hunk 8]
```diff
@@ -36,6 +39,10 @@
 def instance = ET.newInstallation('ECU-TEST', 'C:\\Program Files\\ECU-TEST 8.0')
 instance.startTS()
             </pre>
+            <pre>
+def instance = ET.newInstallation toolName: 'ECU-TEST', installPath: C:\\Program Files\\ECU-TEST 8.0'
+instance.startTS toolLibsIniPath: C:\\ToolLibs.ini, tcpPort: 5017, timeout: '60', keepInstance: false
```
Comment (JanGu-TT) [inline src/main/resources/de/tracetronic/jenkins/plugins/ecutest/tool/StartTSBuilder/help_de.html]: timeout is small integer, would it work with quotas? `'60'`

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/tool/StopETBuilder/help.html [hunk 9]
```diff
@@ -36,6 +39,10 @@
 def instance = ET.newInstallation('ECU-TEST', 'C:\\Program Files\\ECU-TEST 8.0')
 instance.stop()
             </pre>
+            <pre>
+def instance = ET.newInstallation toolName: 'ECU-TEST', installPath: C:\\Program Files\\ECU-TEST 8.0'
+instance.stop timeout: '60'
```
Comment (JanGu-TT) [inline src/main/resources/de/tracetronic/jenkins/plugins/ecutest/tool/StopETBuilder/help.html]: timeout is small integer, would it work with quotas? `'60'`

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/tool/StopETBuilder/help_de.html [hunk 10]
```diff
@@ -36,6 +39,10 @@
 def instance = ET.newInstallation('ECU-TEST', 'C:\\Program Files\\ECU-TEST 8.0')
 instance.stop()
             </pre>
+            <pre>
+def instance = ET.newInstallation toolName: 'ECU-TEST', installPath: C:\\Program Files\\ECU-TEST 8.0'
+instance.stop timeout: '60'
```
Comment (JanGu-TT) [inline src/main/resources/de/tracetronic/jenkins/plugins/ecutest/tool/StopETBuilder/help_de.html]: timeout is small integer, would it work with quotas? `'60'`

File: src/main/resources/de/tracetronic/jenkins/plugins/ecutest/tool/StartTSBuilder/help.html (hunk 7 above)
Comment (cpoenisch) [reply src/main/resources/de/tracetronic/jenkins/plugins/ecutest/tool/StartTSBuilder/help.html]: Timeout property can be both, String or int.
\@DataBoundSetters exist for both types.

**Pull request merged**